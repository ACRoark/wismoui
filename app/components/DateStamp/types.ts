export type DateStampFormat = 'auto' | 'long' | 'short';
