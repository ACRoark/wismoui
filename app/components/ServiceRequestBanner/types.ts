export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
