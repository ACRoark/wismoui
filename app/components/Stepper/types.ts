import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
