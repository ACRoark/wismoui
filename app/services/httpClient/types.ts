export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
