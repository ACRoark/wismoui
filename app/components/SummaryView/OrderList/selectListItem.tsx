import React from 'react';

import { IOrderSearchResult } from 'types';

import OrderListItem from './OrderListItem';
import TransferOrderListItem from './TransferOrderListItem';

const selectListItem = (order: IOrderSearchResult): React.ReactElement => {
  switch (order.orderType) {
    case 'MIMO_START':
    case 'MIMO_STOP':
      return <OrderListItem orderNumber={order.orderNumber} request={order.orderRequests[0]} />;
    case 'MIMO_TRANSFER':
      return <TransferOrderListItem order={order} />;
    default:
      throw new Error('Invalid order type');
  }
};

export default selectListItem;
