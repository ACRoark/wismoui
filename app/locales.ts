export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
