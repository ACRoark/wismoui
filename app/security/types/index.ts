import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
