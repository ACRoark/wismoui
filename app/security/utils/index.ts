import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
