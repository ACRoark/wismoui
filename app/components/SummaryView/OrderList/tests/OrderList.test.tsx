import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { SUPPORTED_LOCALES } from 'locales';
import generate from 'testing/generators';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { IOrderSearchResult } from 'types';

import OrdersList from '../OrderList';
import selectListItem from '../selectListItem';

jest.mock('../selectListItem');

const mockSelectListItem = selectListItem as jest.MockedFunction<typeof selectListItem>;

const emptyList: IOrderSearchResult[] = [];
const oneItemList: IOrderSearchResult[] = generate.orderSearchResults(1, 1);
const twoItemList: IOrderSearchResult[] = generate.orderSearchResults(2, 2);

const expectNoConsoleErrors = (language: string, orders: IOrderSearchResult[]): void => {
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<Router><OrdersList orders={orders} /></Router>, language);

  expect(spy).not.toHaveBeenCalled();
};

const expectComponentToMatchSnapshot = (language: string, orders: IOrderSearchResult[]): void => {
  const tree = createSnapshotWithIntl(<Router><OrdersList orders={orders} /></Router>, language);

  expect(tree).toMatchSnapshot();
};

describe('<OrdersList />', (): void => {
  beforeEach((): void => {
    mockSelectListItem.mockReturnValue(<div>Dummy Item</div>);
  });

  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered with an empty list and the current language is ${language}`, (): void =>
      expectNoConsoleErrors(language, emptyList));

    it(`should not log errors in console when rendered with a single item and the current language is ${language}`, (): void =>
      expectNoConsoleErrors(language, oneItemList));

    it(`should not log errors in console when rendered with multiple items and the current language is ${language}`, (): void =>
      expectNoConsoleErrors(language, twoItemList));

    it(`should render correctly when an empty list is passed and the current language is ${language}`, (): void =>
      expectComponentToMatchSnapshot(language, emptyList));

    it(`should render correctly when a single item is passed and the current language is ${language}`, (): void =>
      expectComponentToMatchSnapshot(language, oneItemList));

    it(`should render correctly when multiple items are passed and the current language is ${language}`, (): void =>
      expectComponentToMatchSnapshot(language, twoItemList));
  });
});
