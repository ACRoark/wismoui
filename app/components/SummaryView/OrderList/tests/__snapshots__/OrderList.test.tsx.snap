// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrdersList /> should render correctly when a single item is passed and the current language is en 1`] = `
<div
  className="dte-wismo-order-list"
>
  <div>
    Dummy Item
  </div>
</div>
`;

exports[`<OrdersList /> should render correctly when a single item is passed and the current language is es 1`] = `
<div
  className="dte-wismo-order-list"
>
  <div>
    Dummy Item
  </div>
</div>
`;

exports[`<OrdersList /> should render correctly when an empty list is passed and the current language is en 1`] = `null`;

exports[`<OrdersList /> should render correctly when an empty list is passed and the current language is es 1`] = `null`;

exports[`<OrdersList /> should render correctly when multiple items are passed and the current language is en 1`] = `
<div
  className="dte-wismo-order-list"
>
  <div>
    Dummy Item
  </div>
  <div>
    Dummy Item
  </div>
</div>
`;

exports[`<OrdersList /> should render correctly when multiple items are passed and the current language is es 1`] = `
<div
  className="dte-wismo-order-list"
>
  <div>
    Dummy Item
  </div>
  <div>
    Dummy Item
  </div>
</div>
`;
